.github/workflows/docker.yml

```diff
@@ -0,0 +1,70 @@
+name: Docker Build & Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      branch:
+        description: skynetlabs/skyd source branch (manual dispatch does NOT push to dockerhub)
+        type: string
+        required: true
+  pull_request:
+  push:
+    tags:
+      - v*
```

Comment:
why not copy our other workflow dispatch for the versioning? 
Are we only allowed to have the one dispath?

Reply:
So the idea is that we publish when we tag the repository and the version of the docker image is the name of the tag. This means we don't have to manually trigger docker image build job and provide a version since that version is already the tag name. It also removes a potential mismatch where we would manually trigger a build with wrong version or version mismatched with what the tag version is. It also ensures that the repo tag version, dockerhub version and skyd tag version is aligned. Let's discuss that on the team sync call today 👍